src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,320 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
             private bool CheckDiagnosticCreation(CompilationAnalysisContext context, SyntaxToken statementIdentifierToken, SyntaxToken keywordIdentifierToken, List<string> ruleNames, SyntaxList<StatementSyntax> statements, ParameterSyntax contextParameter)
             {
                 int statementCount = statements.Count;
-
-                SyntaxToken openParenToken = OpenParenAnalysis(statementIdentifierToken, statements);
-                if (openParenToken.Text == "")
+                if (statementCount < 9)
                 {
-                    IfDiagnostic(context, statements[3], OpenParenIncorrectRule, statementIdentifierToken.Text);
-                    return false;
-                }
-
-                if (statementCount > 4)
-                {
-                    SyntaxToken startToken = StartAnalysis(keywordIdentifierToken, statements);
-                    if (startToken.Text == "")
+                    List<string> diagnosticInfo = DiagnosticCreationCheck(ruleNames, statements[statementCount - 1]);
+                    if (diagnosticInfo.Count == 0 || diagnosticInfo[0] == "" || diagnosticInfo[1] == "")
                     {
-                        IfDiagnostic(context, statements[4], StartSpanIncorrectRule, keywordIdentifierToken.Text);
+                        IfDiagnostic(context, statements[statementCount - 1], DiagnosticIncorrectRule, ruleNames[0]);
```

Comment:
@jepetty But if the if-statement is completely wrong (i.e the wrong type of statement) it doesnt really make sense to just squiggle the condition. That doesnt clearly get across that the entirety of the statement is wrong. I think in the case of it being the wrong type of statement it makes sense to squiggle the entire statement, which is what ReportDiagnostic would do I believe